### ISSUE TITLE
 error: use of undeclared identifier 'getline'; did you mean 'std::getline'?

### ISSUE THREAD

**Issue by yurivict** [surge-synthesizer/vstgui#6]
https://github.com/surge-synthesizer/vstgui/blob/d723171b87f089931815b76f427bcb568d530f2b/vstgui/lib/platform/linux/x11fileselector.cpp#L54

```
/wrkdirs/usr/ports/audio/surge-synthesizer-lv2/work/surge-release_1.6.6-58-g3dbd1be/vstgui.surge/vstgui/lib/platform/linux/x11fileselector.cpp:53:9: error: use of undeclared identifier 'getline'; did you mean 'std::getline'?
                                if (getline (&line, &count, pipe) < 0)
                                    ^~~~~~~
                                    std::getline
/usr/include/c++/v1/istream:1382:1: note: 'std::getline' declared here
getline(basic_istream<_CharT, _Traits>& __is,
^
/wrkdirs/usr/ports/audio/surge-synthesizer-lv2/work/surge-release_1.6.6-58-g3dbd1be/vstgui.surge/vstgui/lib/platform/linux/x11fileselector.cpp:53:9: error: no matching function for call to 'getline'
                                if (getline (&line, &count, pipe) < 0)
                                    ^~~~~~~
/usr/include/c++/v1/istream:1382:1: note: candidate template ignored: could not match 'basic_istream<type-parameter-0-0, type-parameter-0-1>' against 'char **'
getline(basic_istream<_CharT, _Traits>& __is,
^
2 errors generated.
```

Log: http://beefy10.nyi.freebsd.org/data/113i386-default/537862/logs/surge-synthesizer-lv2-1.6.6.58.log

OS: FreeBSD 12.1
Compiler: clang-9

**Comment by baconpaul** [comment]
I don’t think our friends at steinberg meant std::getline since that has a steam signature. I think they meant stdio getline which has the right signature of char** int file*

https://man7.org/linux/man-pages/man3/getdelim.3.html

And then they forgot to include stdio.h which my guess is gets implicitly included in Linux/gcc by one of the headers

Can you try patching in an Include stdio.h in that file and let me know if it works? I don’t have a clang freebsd environment. If it does I am happy to merge that change in.

Thanks!

**Comment by yurivict** [comment]
```#include <stdio.h>``` is definitely missing.
This only fails on the older 11.x systems, I can't verify the fix. But the above include should certainly be added.

**Comment by baconpaul** [comment]
Yeah ok I’ll patch this in when I next update vstgui then - thanks!

**Comment by baconpaul** [comment]
Fixed @ 3cea4e0
